generators/generator-microsoft-bot-adaptive/generators/app/index.js

```diff
@@ -1,155 +1,41 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-const Generator = require('yeoman-generator');
-const path = require('path');
-const { v4: uuidv4 } = require('uuid');
+const rt = require('runtypes');
 const xml2js = require('xml2js');
+const { BaseGenerator, integrations, platforms } = require('../../index');
+const { v4: uuidv4 } = require('uuid');
 
-const INTEGRATION_WEBAPP = 'webapp';
-const INTEGRATION_FUNCTIONS = 'functions';
-
-const PLATFORM_DOTNET = 'dotnet';
-const PLATFORM_JS = 'js';
-const PLATFORM_JAVA = 'java';
-const PLATFORM_PYTHON = 'python';
-
-const PROJECT_TYPEID_WEBAPP = '9A19103F-16F7-4668-BE54-9A1E7A4F7556';
-const PROJECT_TYPEID_FUNCTION = 'FAE04EC0-301F-11D3-BF4B-00C04F79EFBC';
+const options = rt.Record({
```

Comment:
I already liked this when you showed me yesterday, but seeing this again, this is super concise and awesome - love it!